### PR TITLE
Switched from SpeechRecognition to Google Cloud Speech

### LINKED ISSUE #65: Add punctuation to speech-to-text module
Currently there is no punctuation... I think this will pose a problem when trying to generate audio that sounds normal.

### PR DESCRIPTION
**Pull request by andrew-fennell**
This will allow for better STT and added punctuation. We need punctuation to ensure that our TTS flows correctly, so this change is going to be crucial.

The only additional step will be that a Google Cloud credentials file must be populated in order for STT to work. This is not as easy as what we were previously doing, but it will add value to the project.

Closes #65

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -28,6 +28,21 @@ jobs:
...
+        type: ${{ secrets.TYPE }}
+        project_id: ${{ secrets.PROJECT_ID }}
+        private_key_id: ${{ secrets.PRIVATE_KEY_ID }}
+        private_key: ${{ secrets.PRIVATE_KEY }}
+        client_email: ${{ secrets.CLIENT_EMAIL }}
+        client_id: ${{ secrets.CLIENT_ID }}
+        auth_uri: ${{ secrets.AUTH_URI }}
+        token_uri: ${{ secrets.TOKEN_URI }}
+        auth_provider_x509_cert_url: ${{ secrets.AUTH_PROVIDER_X509_CERT_URL }}
+        client_x509_cert_url: ${{ secrets.CLIENT_X509_CERT_URL }}
```
Comment (andrew-fennell) [inline .github/workflows/python-app.yml:45]: This is purely for testing purposes.

File: .gitignore [hunk 2]
```diff
@@ -136,4 +136,7 @@ CogNative/examples/*.wav
 CogNative/examples/saved_embeds/*.ckpt
 
 # Saved models
-CogNative/models/RTVC/saved_models/default/
\ No newline at end of file
+CogNative/models/RTVC/saved_models/default/
+
+# Credentials
+credentials.json
```
Comment (andrew-fennell) [inline .gitignore:142]: To prevent credentials from being uploaded.

File: CogNative/backend/backend.py [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-    translator = speech_transcription()
-
-    file_path = input("Enter your audio file path: ")
-    src_lang = input("Source language: ")
-    dest_lang = input("Destination language: ")
-
-    translated_text = translator.transcribe_audio(file_path, src_lang, dest_lang)
-
-    print(translated_text)
+from .modules import STT, translation
```
Comment (andrew-fennell) [inline CogNative/backend/backend.py:1]: This path needed to be adjusted to import STT and translation from ```main.py```

File: CogNative/backend/modules/STT.py [hunk 4]
```diff
@@ -1,8 +1,16 @@
-import speech_recognition as sr
+import io
+import os
+
+import wave
+from google.cloud import speech
```
Comment (andrew-fennell) [inline CogNative/backend/modules/STT.py:5]: ```speech_recognition``` -> ```google-cloud-speech```

**Review by andrew-fennell — COMMENTED** [review]
Added comments to help reviewer.

**Review by JacobSmith1020 — APPROVED** [review]
Everything working locally :) Same small issues in PR67 may need to be addressed at some point.

**Pull request merged**